src/topology/uniform_space/uniform_convergence.lean

```diff
@@ -251,6 +306,40 @@ begin
   exact ⟨F m x, ⟨hm', htsymm (hm x hx)⟩⟩,
 end
 
+lemma uniform_cauchy_seq_on.mono {s' : set α} (hf : uniform_cauchy_seq_on F p s) (hss' : s' ⊆ s) :
+  uniform_cauchy_seq_on F p s' :=
+λ u hu, (hf u hu).mono (λ x hx y hy, hx y (hss' hy))
+
+/-- Composing on the right by a function preserves uniform convergence -/
+lemma uniform_cauchy_seq_on.comp {γ : Type*} (hf : uniform_cauchy_seq_on F p s) (g : γ → α) :
+  uniform_cauchy_seq_on (λ n, F n ∘ g) p (g ⁻¹' s) :=
+λ u hu, (hf u hu).mono (λ x hx y hy, hx (g y) hy)
+
+/-- Composing on the left by a uniformly continuous function preserves
+uniform convergence -/
+lemma uniform_cauchy_seq_on.comp' [uniform_space γ] {g : β → γ} (hf : uniform_cauchy_seq_on F p s)
+  (hg : uniform_continuous g) :
+  uniform_cauchy_seq_on (λ n, g ∘ (F n)) p s :=
+λ u hu, hf _ (hg hu)
+
+lemma uniform_cauchy_seq_on.prod' {β' : Type*} [uniform_space β'] {F' : ι → α → β'}
```

Comment:
```suggestion
lemma uniform_cauchy_seq_on.prod_mk {β' : Type*} [uniform_space β'] {F' : ι → α → β'}
```

Reply:
You mean the unprimed version should be something like this?
```lean
lemma uniform_cauchy_seq_on.prod {ι' β' : Type*} [uniform_space β'] {p' : filter ι'}
  {F' : ι' → α → β'}
  (h : uniform_cauchy_seq_on F p s) (h' : uniform_cauchy_seq_on F' p' s) :
  uniform_cauchy_seq_on (λ (i : ι × ι') a, (F i.1 a, F' i.2 a)) (p ×ᶠ p') s :=
sorry
```
Can you conclude you version from this version (maybe using that `λ i, (i, i)` tends to `at_top` if `i` tends to `at_top`)? 
If so, that would also be useful to have.

Reply:
Seeing what I can do here

Reply:
OK, so I have produced the relevant proof, but it involves oddly hinting a proposition to the compiler. Do you know of a better way to do this kind of hinting?

Reply:
OK, I've filled in the missing `tendsto` as well. So this is "complete" but it is... ugly. The issue comes down to needing to compose a bunch of associativity and swap maps and facts about tendsto. If there's a better way to do this, then I'm happy to implement!